File: 746. Min Cost Climbing Stairs/746.py
class Solution:
    def minCostClimbingStairs(self, cost: list) -> int:
        for i in range(2, len(cost)):
            cost[i] = min(cost[i-1], cost[i-2]) + cost[i]
        return min(cost[-1],cost[-2])

    def func(self, cost):
        pre, cur = cost[0], cost[1]
        for i in range(2, len(cost)):
            pre, cur = cur, min(pre, cur) + cost[i]
        return min(pre,cur)

a = Solution()
print(a.minCostClimbingStairs([1, 100, 1, 1, 1, 100, 1, 1, 100, 1]))

File: 746. Min Cost Climbing Stairs/746.go
package main

import "fmt"

func main(){
	fmt.Println(minCostClimbingStairs([]int{1, 100}))
}

func minCostClimbingStairs(cost []int) int {
    for index:=2; index<len(cost); index++{
		if cost[index-1] > cost[index-2]{
			cost[index] = cost[index-2] + cost[index]
		}else{
			cost[index] = cost[index-1] + cost[index]
		}
	}
	if cost[len(cost)-1]<cost[len(cost)-2]{
		return cost[len(cost)-1]
	}
	return cost[len(cost)-2]
}